### PR TITLE
BACK-1607-Save TxViews in Interpreter

### PR DESCRIPTION
**Pull request by estrauser-ledger**
## What is this about?

Save transactions as TransactionView in Interpreter

## Cute picture of lama
lama bg

### CODE REVIEW

File: coins/bitcoin/common/src/main/scala/co/ledger/lama/bitcoin/common/clients/grpc/mocks/InterpreterClientMock.scala [hunk 1]
```diff
@@ -63,14 +57,20 @@ class InterpreterClientMock extends InterpreterClient {
...
+      .collect { case TransactionView(_, _, _, _, _, _, _, Some(block), _) =>
+        BlockView(
+          block.hash,
+          block.height,
+          block.time
+        )
+      }
       .distinct
       .sortBy(_.height)
       .reverse
```
Comment (alekece) [inline coins/bitcoin/common/src/main/scala/co/ledger/lama/bitcoin/common/clients/grpc/mocks/InterpreterClientMock.scala:71]: Would it be more efficient to sort like this (we remove a total iteration on the list) ?
```suggestion
      .sortBy(-_.height)
```

File: coins/bitcoin/protobuf/src/main/protobuf/interpreter.proto [hunk 2]
```diff
@@ -31,15 +31,15 @@ enum SortingOrder {
 
 message SaveTransactionsRequest {
   bytes account_id = 1;
-  repeated Transaction transactions = 2;
+  repeated TransactionView transactions = 2;
```
Comment (mbasire-ledger) [inline coins/bitcoin/protobuf/src/main/protobuf/interpreter.proto:34]: That was not mandatory to rename Transaction here. BTW, it is not a problem.

**Review by mbasire-ledger — COMMENTED** [review]
Good news to get rid off of the model of the explorer. I hope we suppress the View suffix later !

**Review by nukky94 — APPROVED** [review]

File: coins/bitcoin/protobuf/src/main/protobuf/interpreter.proto (hunk 2 above)
Comment (estrauser-ledger) [reply coins/bitcoin/protobuf/src/main/protobuf/interpreter.proto:34]: Yep we'll rename it later, after the refacto

**Pull request merged**